sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteRename.java

```diff
@@ -232,7 +238,7 @@ private PendingJobData startWriteRename(
     String bqLocation =
         BigQueryHelpers.getDatasetLocation(datasetService, ref.getProjectId(), ref.getDatasetId());
 
-    String projectId = ref.getProjectId();
+    String projectId = loadJobProjectId == null ? ref.getProjectId() : loadJobProjectId.get();
```

Comment:
Should this be something like the following?
```suggestion
    String projectId = loadJobProjectId == null || loadJobProjectId.get() == null ? ref.getProjectId() : loadJobProjectId.get();
```

Reply:
should we include the additional check here as well?  https://github.com/apache/beam/blob/92ba229336aa9cc4132f498d7049ecf67bc1c3b3/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteTables.java#L491

 I wasn't sure if  `ValueProvider<String>` accepts nulls as well.

Reply:
Yes, please. At least the default value is nullable: https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java#L206.  And the underlying implementation (https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java#L267) of RuntimeValueProvider is a StaticValueProvider that allows nullable values: https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java#L85

Reply:
added!